### DISCUSSION TITLE
구조적 변경이 필요한 부분

### DISCUSSION THREAD

**Discussion by sm-amoled** [cau-bootcamp/wehavit#181] [General]
리펙토링을 진행하면서 현재 기능의 개선이 필요한 부분들이 많이 보인다. 얼른 고쳐봐야지 😘

**Comment by sm-amoled** [comment]
Reaction을 불러오는 방식의 변경이 필요함

현재 방식 : 가장 최근의 포스트에서 아직 보지 않은 반응들을 가져와 보여주기
단점 
- 하루에 여러 목표에 대한 포스트를 따로 올리는 구조에서는 적용이 불가능함 (거의 확정된 기능 개편이므로 ㅎㅎ)
- 여러 포스트에 남겨진 목표를 확인하지 못하게 되는 구조이므로, 수정이 필요함

개선 방식 제안
- 나에게 남겨진 반응 중 내가 아직 보지 못한 반응들을 따로 기록해두기
  1. 포스트 아래에 반응이 남겨짐 (파이어베이스 상으로는 포스트 하위의 리액션 컬렉션에 담김)
  2. 동시에, 받는 사람의 고유 reaction 보관함에도 표시 남기기 (user/$userid/received_reactions/여기에 담기)
  3. 아마 reaction 보관함에는 [post id / reaction id] 만 남기고, 읽으면 보관함에서 지우기 (한 번만 받을 수 있음)
  4. 각 포스트에 남겨진 리액션을 다시 보고싶으면 포스트 단위로 리액션을 확인할 수 있게 하면 되겠다.
- 요런 방식으로 구조를 변경해볼까 싶음!

**Reply by sm-amoled** [reply]
변경 완료함! 아래 PR 참고.
- #187

**Reply by IamGroooooot** [reply]
와웅 수고했어용. 이제 저도 합류합니닷
